fir/src/org/jetbrains/kotlin/idea/fir/applicators/AddArgumentNamesApplicators.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+package org.jetbrains.kotlin.idea.fir.applicators
+
+import com.intellij.psi.PsiComment
+import org.jetbrains.kotlin.idea.KotlinBundle
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicatorInput
+import org.jetbrains.kotlin.idea.api.applicator.applicator
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.KtCallElement
+import org.jetbrains.kotlin.psi.KtLambdaArgument
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import org.jetbrains.kotlin.psi.KtValueArgument
+import org.jetbrains.kotlin.psi.psiUtil.getPrevSiblingIgnoringWhitespace
+
+object AddArgumentNamesApplicators {
+    class SingleArgumentInput(val name: Name) : HLApplicatorInput
+
+    class MultipleArgumentsInput(val argumentNames: Map<KtValueArgument, Name>) : HLApplicatorInput
+
+    val singleArgumentApplicator = applicator<KtValueArgument, SingleArgumentInput> {
+        familyName(KotlinBundle.lazyMessage("add.name.to.argument"))
+
+        actionName { _, input -> KotlinBundle.message("add.0.to.argument", input.name)  }
+
+        isApplicableByPsi { element ->
+            // Not applicable when lambda is trailing lambda after argument list (e.g., `run {  }`); element is a KtLambdaArgument.
+            // Note: IS applicable when lambda is inside an argument list (e.g., `run({  })`); element is a KtValueArgument in this case.
```

Comment:
Hm, then... should we rather check if `element is KtValueArgument` ?

Reply:
`KtLambdaArgument` is a subclass of `KtValueArgument` so we can't use that.